src/main/drivers/sonar_hcsr04.c

```diff
@@ -75,9 +75,13 @@ void EXTI9_5_IRQHandler(void)
 }
 #endif
 
-void hcsr04_init(const sonarHardware_t *initialSonarHardware, sonarRange_t *sonarRange)
+void hcsr04_set_sonar_hardware(const sonarHardware_t *initialSonarHardware)
```

Comment:
Yes.

It is to do with detection of the sonar device. There is no way to detect if the user has attached an HCSR04, and the SRF10 can only be detected once the I2C bus has been initialised, which is after `pwmInit` has been called, and `sonarConfigureHardware` must be called before `pwmInit`, since the HCSR04 may be attached to the PWM output pins.

So the logic used is as follows:

1. if the user has set the sonar feature, assume they have actually attached a sonar device.
2. at the time of `pwmInit` make the assumption that this is an HCSR04 since we cannot detect either an HCSR04 (since there is no means of detecting it) or an SRF10 (since the I2C bus has not been initialised)
3. `sonarInit` is called after the I2C bus has been initialised, so it can detect the presence of an SRF10. If one is found, then use it. If no SRF10 is found then assume the user has attached an HCSR04.